GVRf/Sample/gvrbulletsample/src/org/gearvrf/bulletsample/BulletSampleViewManager.java

```diff
@@ -0,0 +1,225 @@
+				new Vector3(0.0f, 0.0f, 0.0f));
+		MotionState floorState = new MotionState();
+		mBullet.createAndAddRigidBody(floorGeometry, floorState);
+
+		/*
+		 * Create Some cubes in Bullet world and hit it with a sphere
+		 */
+		addCube(scene, 0.0f, 1.0f, -9.0f, CUBE_MASS);
+		addCube(scene, 0.0f, 1.0f, -10.0f, CUBE_MASS);
+		addCube(scene, 0.0f, 1.0f, -11.0f, CUBE_MASS);
+		addCube(scene, 1.0f, 1.0f, -9.0f, CUBE_MASS);
+		addCube(scene, 1.0f, 1.0f, -10.0f, CUBE_MASS);
+		addCube(scene, 1.0f, 1.0f, -11.0f, CUBE_MASS);
+		addCube(scene, 2.0f, 1.0f, -9.0f, CUBE_MASS);
+		addCube(scene, 2.0f, 1.0f, -10.0f, CUBE_MASS);
+		addCube(scene, 2.0f, 1.0f, -11.0f, CUBE_MASS);
+
+		addCube(scene, 0.0f, 2.0f, -9.0f, CUBE_MASS);
+		addCube(scene, 0.0f, 2.0f, -10.0f, CUBE_MASS);
+		addCube(scene, 0.0f, 2.0f, -11.0f, CUBE_MASS);
+		addCube(scene, 1.0f, 2.0f, -9.0f, CUBE_MASS);
+		addCube(scene, 1.0f, 2.0f, -10.0f, CUBE_MASS);
+		addCube(scene, 1.0f, 2.0f, -11.0f, CUBE_MASS);
+		addCube(scene, 2.0f, 2.0f, -9.0f, CUBE_MASS);
+		addCube(scene, 2.0f, 2.0f, -10.0f, CUBE_MASS);
+		addCube(scene, 2.0f, 2.0f, -11.0f, CUBE_MASS);
+
+		addCube(scene, 0.0f, 3.0f, -9.0f, CUBE_MASS);
+		addCube(scene, 0.0f, 3.0f, -10.0f, CUBE_MASS);
+		addCube(scene, 0.0f, 3.0f, -11.0f, CUBE_MASS);
+		addCube(scene, 1.0f, 3.0f, -9.0f, CUBE_MASS);
+		addCube(scene, 1.0f, 3.0f, -10.0f, CUBE_MASS);
+		addCube(scene, 1.0f, 3.0f, -11.0f, CUBE_MASS);
+		addCube(scene, 2.0f, 3.0f, -9.0f, CUBE_MASS);
+		addCube(scene, 2.0f, 3.0f, -10.0f, CUBE_MASS);
+		addCube(scene, 2.0f, 3.0f, -11.0f, CUBE_MASS);
+
+		/*
+		 * Throw a sphere from top
+		 */
+		addSphere(scene, 1.0f, 1.5f, 100.0f, -10.0f, 20.0f);
+	}
+
+	@Override
+	public void onStep() {
+		mBullet.doSimulation(1.0f / 60.0f, 10);
+		for (RigidBody body : rigidBodiesSceneMap.keySet()) {
+			if (body.geometry.shape.getType() == ShapeType.SPHERE_SHAPE_PROXYTYPE) {
+				rigidBodiesSceneMap
+						.get(body)
+						.getTransform()
+						.setPosition(
+								body.motionState.resultSimulation.originPoint.x,
+								body.motionState.resultSimulation.originPoint.y,
+								body.motionState.resultSimulation.originPoint.z);
+			}
+
+			if (body.geometry.shape.getType() == ShapeType.BOX_SHAPE_PROXYTYPE) {
+				rigidBodiesSceneMap
+						.get(body)
+						.getTransform()
+						.setPosition(
+								body.motionState.resultSimulation.originPoint.x,
+								body.motionState.resultSimulation.originPoint.y,
+								body.motionState.resultSimulation.originPoint.z);
+			}
+		}
+	}
+
+	private GVRSceneObject quadWithTexture(float width, float height,
+			String texture) {
+		FutureWrapper<GVRMesh> futureMesh = new FutureWrapper<GVRMesh>(
+				mGVRContext.createQuad(width, height));
+		GVRSceneObject object = null;
+		try {
+			object = new GVRSceneObject(mGVRContext, futureMesh,
+					mGVRContext.loadFutureTexture(new GVRAndroidResource(
+							mGVRActivity, texture)));
+		} catch (IOException e) {
```

Comment:
Normally we uses GVRContext to do with GVRAndroidResource not GVRActivity. Just a minor issue here. Actually this is the only place that you uses mGVRActivity. Probably you actually don't need this in this sample.